pkg/fleet/installer/setup/djm/emr.go

```diff
@@ -0,0 +1,186 @@
+	ReleaseLabel string `json:"releaseLabel"`
+}
+
+// SetupEmr sets up the DJM environment on EMR
+func SetupEmr(s *common.Setup) error {
+
+	s.Packages.Install(common.DatadogAgentPackage, emrAgentVersion)
+	s.Packages.Install(common.DatadogAPMInjectPackage, emrInjectorVersion)
+	s.Packages.Install(common.DatadogAPMLibraryJavaPackage, emrJavaTracerVersion)
+
+	os.Setenv("DD_APM_INSTRUMENTATION_ENABLED", "host")
+
+	hostname, err := os.Hostname()
+	if err != nil {
+		return fmt.Errorf("failed to get hostname: %w", err)
+	}
+	s.Config.DatadogYAML.Hostname = hostname
+	s.Config.DatadogYAML.DJM.Enabled = true
+	s.Config.InjectTracerYAML.AdditionalEnvironmentVariables = tracerEnvConfigEmr
+
+	// Ensure tags are always attached with the metrics
+	s.Config.DatadogYAML.ExpectedTagsDuration = "10m"
+	isMaster, clusterName, err := setupCommonEmrHostTags(s)
+	if err != nil {
+		return fmt.Errorf("failed to set tags: %w", err)
+	}
+	if isMaster {
+		setupEmrResourceManager(s, clusterName)
+	}
+	return nil
+}
+
+func setupCommonEmrHostTags(s *common.Setup) (bool, string, error) {
+
+	instanceInfoRaw, err := os.ReadFile(filepath.Join(emrInfoPath, "instance.json"))
+	if err != nil {
+		return false, "", fmt.Errorf("error reading instance file: %w", err)
+	}
+
+	var info emrInstanceInfo
+	if err = json.Unmarshal(instanceInfoRaw, &info); err != nil {
+		return false, "", fmt.Errorf("error umarshalling instance file: %w", err)
+	}
+
+	setHostTag(s, "instance_group_id", info.InstanceGroupID)
+	setHostTag(s, "is_master_node", strconv.FormatBool(info.IsMaster))
+	s.Span.SetTag("host_tag."+"is_master_node", info.IsMaster)
+
+	extraInstanceInfoRaw, err := os.ReadFile(filepath.Join(emrInfoPath, "extraInstanceData.json"))
+	if err != nil {
+		return info.IsMaster, "", fmt.Errorf("error reading extra instance data file: %w", err)
+	}
+
+	var extraInfo extraEmrInstanceInfo
+	if err = json.Unmarshal(extraInstanceInfoRaw, &extraInfo); err != nil {
+		return info.IsMaster, "", fmt.Errorf("error umarshalling extra instance data file: %w", err)
+	}
+	setHostTag(s, "job_flow_id", extraInfo.JobFlowID)
+	setHostTag(s, "cluster_id", extraInfo.JobFlowID)
+	setHostTag(s, "emr_version", extraInfo.ReleaseLabel)
+	s.Span.SetTag("emr_version", extraInfo.ReleaseLabel)
+
+	emrResponseRaw, err := executeCommandWithTimeout("aws", "emr", "describe-cluster", "--cluster-id", extraInfo.JobFlowID)
+	if err != nil {
+		log.Warn("error describing emr cluster, using cluster id as name")
+		return info.IsMaster, extraInfo.JobFlowID, nil
+	}
+	var response emrResponse
+	if err = json.Unmarshal(emrResponseRaw, &response); err != nil {
+		return info.IsMaster, extraInfo.JobFlowID, fmt.Errorf("error unmarshalling AWS EMR response,  using cluster id as name: %w", err)
+	}
+	clusterName := response.Cluster.Name
+	if clusterName == "" {
+		log.Warn("clusterName is empty, using cluster id as name")
+		clusterName = extraInfo.JobFlowID
+	}
+	setHostTag(s, "cluster_name", clusterName)
```

Comment:
Let's never return an error if anything fails on the `describe-cluster` and just use the cluster id as a fallback for the name

Since this operation can be a source of error, i think we should create a span for it

Reply:
Thanks for the good idea. I addressed it by isolating the command execution in a separate function and having spans for it so the error is easily trackable.